lighthouse-core/audits/service-worker.js

```diff
@@ -73,23 +73,33 @@ class ServiceWorker extends Audit {
    * @param {Array<LH.Crdp.ServiceWorker.ServiceWorkerVersion>} matchingSWVersions
    * @param {Array<LH.Crdp.ServiceWorker.ServiceWorkerRegistration>} registrations
    * @param {URL} pageUrl
-   * @return {string|undefined}
+   * @return {{scopeUrl: string; scriptUrl: string} | undefined}
    */
-  static getControllingScopeUrl(matchingSWVersions, registrations, pageUrl) {
+  static getControllingServiceWorker(matchingSWVersions, registrations, pageUrl) {
     // Find the normalized scope URLs of possibly-controlling SWs.
-    const matchingScopeUrls = matchingSWVersions
-      .map(v => registrations.find(r => r.registrationId === v.registrationId))
-      .filter(/** @return {r is LH.Crdp.ServiceWorker.ServiceWorkerRegistration} */ r => !!r)
-      .map(r => new URL(r.scopeURL).href);
+    /** @type {{ scopeUrl: string; scriptUrl: string; }[]} */
```

Comment:
nit: prefer `Array<...>`, especially for complex types

there's a bit of an unofficial war between me, @patrickhulce and @brendankenny about when to use `Array<...>` vs `...[]` :) I prefer using `[]` for simple, non-compound types like `string[]` or `LH.Artifact.Blah[]`, but others prefer always using it. eventually you may pick a side ...

Reply:
for the record I also have absolutely no problem with `string[]` or `number[]` :) 

I think we all agree when the types grow to be more complex (I'd probably switch earlier than @connorjclark and use `Array<LH.Nested.Thing>`) the `Array<T>` form is easier to read 👍